lib/rules/jsx-quotes.js

```diff
@@ -19,11 +21,19 @@ var astUtils = require("../ast-utils");
 var QUOTE_SETTINGS = {
     "prefer-double": {
         quote: "\"",
-        description: "singlequote"
+        alternateQuote: "'",
+        description: "singlequote",
+        convert: function(str) {
+            return toDoubleQuotes(str);
```

Comment:
@ianks you could use something similar to https://github.com/eslint/eslint/blob/master/lib/rules/quotes.js#L48 here
```js
convert = function(str) {
    var newQuote = this.quote;
    var oldQuote = str[0];
    if (newQuote === oldQuote) {
        return str;
    }
    return newQuote + str.slice(1, -1).replace(/(&#39;)|(&#34;|&quot;)|['"]/g, function(match, q1, q2) {
        if (match === newQuote) {
            return "&#" + match.charCodeAt(0) + ";"
        }
        if (oldQuote === "'" && q1 || oldQuote === "\"" && q2) {
            return oldQuote;
        }
        return match;
    }) + newQuote;
};
```